### PR TITLE
APS-1321: Remove references to genderForAp

### PR DESCRIPTION
**Pull request by froddd**
# Context

https://dsdmoj.atlassian.net/browse/APS-1321

# Changes in this PR

Removes the use of `Application.genderForAp`, converting any logic to using `Application.isWomensApplication` instead. Refactor of Task. utilities tests.

_Note: for now the wording has not been changed, so the task allocation screen shows 'Gender for AP: Male/Female' This will need to be updated._

# To-do

- [x] Update wording

## Screenshots of UI changes

n/a

### CODE REVIEW

File: server/utils/tasks/index.test.ts [hunk 1]
```diff
@@ -87,201 +71,116 @@ describe('index', () => {
...
-          },
-        ])
+    describe('when the application contains an arrival date', () => {
+      beforeEach(() => {
+        ;(arrivalDateFromApplication as jest.Mock).mockReturnValue('2022-01-01')
+      })
+
+      it('renders the arrival date', () => {
+        expect(taskSummary(task, application)).toEqual(
+          expect.arrayContaining([
```
Comment (bobmeredith) [inline server/utils/tasks/index.test.ts:113]: nice!

File: server/utils/tasks/index.test.ts [hunk 2]
```diff
@@ -87,201 +71,116 @@ describe('index', () => {
...
+            value: { text: 'Unallocated' },
           },
           {
             key: { text: 'Gender for AP' },
-            value: { text: sentenceCase(application.genderForAp) },
-          },
-          {
-            key: { text: 'Applicant PDU' },
-            value: { text: 'test' },
+            value: { text: 'Male' },
```
Comment (bobmeredith) [inline server/utils/tasks/index.test.ts:83]: I'm thinking gender should be man/woman or perhaps for APs, `men's` and `women's`

**Review by bobmeredith — COMMENTED** [review]
Looks good.
One query about gender terms.

**Review by bobmeredith — APPROVED** [review]
Apologies - You said in the description the terms still had to be updated.
LGTM

File: server/utils/tasks/index.ts [hunk 3]
```diff
@@ -116,26 +115,24 @@ const taskSummary = (task: Task, application: Application): Array<SummaryListIte
...
 
-  if (application.genderForAp) {
-    summary.push({
-      key: { text: 'Gender for AP' },
-      value: { text: sentenceCase(application.genderForAp) },
-    })
-  }
+  summary.push({
+    key: { text: 'Gender for AP' },
+    value: { text: application.isWomensApplication ? 'Female' : 'Male' },
```
[the remark this answers is not in the document]
Comment (froddd) [reply server/utils/tasks/index.ts:134]: Yes -- that's the to-do item in the description. The wording for the whole summary item is likely to change, but I thought the logic and refactor could be reviewed ahead of that!

**Comment by bobmeredith** [comment]
Content changes look good to me

**Pull request merged**